### PR TITLE
fix: publisher info

### PR DESCRIPTION
**Pull request by elrrrrrrr**
> Fixed an issue with the `_npmUser` field setting during the initial sync.
1. 🧶 Fixed _npmUser field setting issue during initial sync, should use displayName.
2. 🧶 skip diff abbreviated meta for _npmUser
3. 🤖 Refined some TypeScript definitions, use isEqual to diff metas

-------

> 修复首次同步时，`_npmUser` 字段设置异常
1. 🧶 修复 publisher 匹配，获取逻辑，应当用 displayName 进行匹配
2. 🧶 精简 meta 信息，跳过比较不存在的 _npmUser 字段
2. 🤖 调整部分 ts 定义，使用 isEqual 来进行 diff 比较

### CODE REVIEW

File: app/core/service/PackageManagerService.ts [hunk 1]
```diff
@@ -53,7 +53,7 @@ export interface PublishPackageCmd {
   // name don't include scope
   name: string;
   version: string;
-  description: string;
+  description?: string;
   packageJson: PackageJSONType;
```
Comment (gemwuu) [inline app/core/service/PackageManagerService.ts:57]: is it possible to use `type-fest`.`PackageJson`?

Comment (elrrrrrrr) [reply app/core/service/PackageManagerService.ts:57]: 会有一些私有字段，PackageJSONType 会做一些 patch

**Review by fengmk2 — APPROVED** [review]
+1

File: app/core/service/PackageSyncerService.ts [hunk 2]
```diff
@@ -607,7 +607,8 @@ export class PackageSyncerService extends AbstractService {
         // need libc field https://github.com/cnpm/cnpmcore/issues/187
         // fix _npmUser field since https://github.com/cnpm/cnpmcore/issues/553
         const metaDataKeys = [ 'peerDependenciesMeta', 'os', 'cpu', 'libc', 'workspaces', 'hasInstallScript', 'deprecated', '_npmUser' ];
-        let diffMeta: any;
+        const ignoreInAbbreviated = [ '_npmUser' ];
```
Comment (fengmk2) [inline app/core/service/PackageSyncerService.ts:610]: @elrrrrrrr 想不起来为什么这里需要忽略 _npmUser 了

**Pull request merged**